webrecorder/browsertrix
# Require current password to reset password in account settings

Fixes #1269 

Implements a forked custom user router to disable password changes via `PATCH /users/me` and `PATCH /users/<id>`, since overriding fastapi endpoints is not straightforward (if an endpoint is redefined after being initially defined, the OpenAPI docs change but fastapi continues to use the first defined route method). The fork is copy and pasted from fastapi-users 9.2.2 with a few changes.

Follow-up PR #1281 then extends this to the auth router, for custom changes to prevent brute force password cracking.

A new `PUT /users/me/password-change` endpoint is implemented as recommended in the issue, with the following two exceptions stemming from user_manager internals restrictions:
- the new password should be passed in as `password`, not new
- the body must include the user's email in the body as `email`

The body for `PUT /users/me/password-change` should be:
```json
{
    "current": "currentpassword",
    "password": "newpassword",
    "email": "useremail@example.com"
}
```
Also updates the frontend's account settings to account for the backend changes.

## To test

1. Run local instance from this branch
2. Log in to frontend
3. Click on user icon in upper-right, then "Account Settings"
4. Attempt changing password with invalid current password and verify it fails
5. Attempt changing password with valid current password and ensure it succeeds
